src/governance-classes.cpp

```diff
@@ -0,0 +1,492 @@
+                      << endl; );
+            continue;
+        }
+
+        // MAKE SURE THIS TRIGGER IS ACTIVE VIA FUNDING CACHE FLAG
+
+        if(pObj->fCachedFunding)  {
+            DBG( cout << "IsSuperblockTriggered returning true" << endl; );
+            return true;
+        }       
+        else  {
+            DBG( cout << "IsSuperblockTriggered No fCachedFunding, continuing" << endl; );
+        }
+    }
+
+    return false;
+}
+
+
+bool CSuperblockManager::GetBestSuperblock(CSuperblock_sptr& pBlock, int nBlockHeight)
+{
+    AssertLockHeld(governance.cs);
+    std::vector<CSuperblock_sptr> vecTriggers = triggerman.GetActiveTriggers();
+    int nYesCount = 0;
+
+    BOOST_FOREACH(CSuperblock_sptr superblock, vecTriggers)
+    {
+        if(!superblock)  {
+            DBG( cout << "GetBestSuperblock Not a superblock, continuing" << endl; );
+            continue;
+        }
+
+        CGovernanceObject* pObj = superblock->GetGovernanceObject();
+
+        if(!pObj)  {
+            DBG( cout << "GetBestSuperblock pObj is NULL, continuing" << endl; );
+            continue;
+        }
+
+        if(nBlockHeight != superblock->GetBlockStart()) {
+            DBG( cout << "GetBestSuperblock Not the target block, continuing" << endl; );
+            continue;
+        }
+        
+        // DO WE HAVE A NEW WINNER?
+
+        int nTempYesCount = pObj->GetAbsoluteYesCount(VOTE_SIGNAL_FUNDING);
+        DBG( cout << "GetBestSuperblock nTempYesCount = " << nTempYesCount << endl; );
+        if(nTempYesCount > nYesCount)  {
+            nYesCount = nTempYesCount;
+            pBlock = superblock;
+            DBG( cout << "GetBestSuperblock Valid superblock found, pBlock set" << endl; );
+        }       
+    }
+
+    return nYesCount > 0;
+}
+
+/**
+*   Create Superblock Payments
+*
+*   - Create the correct payment structure for a given superblock
+*/
+
+void CSuperblockManager::CreateSuperblock(CMutableTransaction& txNew, CAmount nFees, int nBlockHeight)
+{
+    DBG( cout << "CSuperblockManager::CreateSuperblock Start" << endl; );
+
+    LOCK(governance.cs);
+    AssertLockHeld(cs_main);
+
+    if(!chainActive.Tip()) {
+        DBG( cout << "CSuperblockManager::CreateSuperblock No active tip, returning" << endl; );
+        return;
+    }
+
+    // GET THE BEST SUPERBLOCK FOR THIS BLOCK HEIGHT
+
+    CSuperblock_sptr pBlock;
+    if(!CSuperblockManager::GetBestSuperblock(pBlock, nBlockHeight))  {
+        LogPrint("superblock", "CSuperblockManager::CreateSuperblock: Can't find superblock for height %d\n", nBlockHeight);
+        DBG( cout << "CSuperblockManager::CreateSuperblock Failed to get superblock for height, returning" << endl; );
+        return;
+    }
+
+    // CONFIGURE SUPERBLOCK OUTPUTS 
+
+    DBG( cout << "CSuperblockManager::CreateSuperblock Number payments: " << pBlock->CountPayments() << endl; );
+
+    txNew.vout.resize(pBlock->CountPayments());
+    for(int i = 0; i <= pBlock->CountPayments(); i++)  {
```

Comment:
Given the fact that `CountPayments` is essentially `vecPayments.size()` is it ok to have `<=` here? Looks like we are going out of bounds and should be `<`.